test/incr_test.ml

```diff
@@ -270,6 +270,247 @@ let bind_test () =
     destroy_comp weird_not_computation
   end
 
+let internals_test () =
+  begin
+    let ops = ref 0 in
+    let reset_ops () = ops := 0 in
+    let rec sum_range ~lo ~hi xs =
+      delay @@ fun () ->
+      if hi - lo <= 1 then begin
+        xs.(lo)
+      end
+      else
+        let mid = lo + ((hi - lo) asr 1) in
+        map2
+          ~fn:(fun x y ->
+            incr ops;
+            Int.add x y)
+          (sum_range ~lo ~hi:mid xs) (sum_range ~lo:mid ~hi xs)
+    in
+    let arr = Array.of_list [1; 2; 3; 4; 5; 6; 7; 8; 9; 10] in
+    let var_arr = Array.map Var.create arr in
+    let t_arr = Array.map Var.watch var_arr in
+    let sum = sum_range ~lo:0 ~hi:(Array.length t_arr) t_arr in
+    let executor =
+      {
+        run = (fun f -> f ());
```

Comment:
BTW, I often format records, especially records with functions, such that I first bind the values with the same names as the record fields and then use the record punning notation:

```ocaml
let executor =
  let run f = f ()
  and par_do l r =
    let lres = l () in
    (lres, r ())
  in
  { run; par_do }
```

This often formats to fewer lines and the syntax for functions can be nicer.